packages/shared/lib/clients/orchestrator.ts

```diff
@@ -669,6 +677,53 @@ export class Orchestrator {
         }
     }
 
+    async updateSyncFrequency({
+        syncId,
+        interval,
+        syncName,
+        environmentId,
+        activityLogId,
+        logCtx
+    }: {
+        syncId: string;
+        interval: string;
+        syncName: string;
+        environmentId: number;
+        activityLogId?: number;
+        logCtx?: LogContext;
+    }): Promise<Result<void>> {
+        const isGloballyEnabled = await featureFlags.isEnabled('orchestrator:schedule', 'global', false);
+        const isEnvEnabled = await featureFlags.isEnabled('orchestrator:schedule', `${environmentId}`, false);
+        const isOrchestrator = isGloballyEnabled || isEnvEnabled;
```

Comment:
we do update in both temporal and orchestrator and return the result based on the flag. The idea is to have the schedules mirrored in both systems and have their states match. The same flag will be used in `jobs` determine which execution between the temporal one and the orchestrator one will actually run.